compositor_render/src/transformations/web_renderer/packet.rs

```diff
@@ -0,0 +1,56 @@
+use std::{
+    io::{Read, Write},
+    net::TcpStream,
+};
+
+pub struct Packet<'a>(pub &'a [u8]);
+
+impl<'a> Packet<'a> {
+    pub fn send(&self, stream: &mut TcpStream) -> Result<(), PacketError> {
+        let header = (self.0.len() as u32).to_be_bytes();
+        let buffer = [&header, self.0].concat();
+        stream.write(&buffer).map_err(PacketError::SendFailure)?;
+
+        Ok(())
+    }
+
+    pub fn read(stream: &mut TcpStream) -> Result<Vec<u8>, PacketError> {
+        let mut len = -1;
+        let mut header = Vec::with_capacity(4);
+        let mut msg = Vec::new();
+        let mut buf = [0u8; 65535];
```

Comment:
Correct me if I'm wrong, but I think this will allocate this buffer on the stack. It should be on the heap, it's too large for the stack, use vector instead.

Reply:
As far as I know it should fit on the stack but I changed the method's implementation anyway